test/Index/index_only_sdk_system_modules.swift

```diff
@@ -0,0 +1,58 @@
+// RUN: %empty-directory(%t)
+// RUN: %empty-directory(%t/cache)
+// RUN: %empty-directory(%t/idx)
+// RUN: split-file %s %t
+
+/// Create a Swift module to import.
+// RUN: %target-swift-frontend -emit-module \
+// RUN:   -enable-library-evolution -swift-version 5 \
+// RUN:   -emit-module-path %t/LocalSystemModule.swiftmodule \
+// RUN:   -emit-module-interface-path %t/LocalSystemModule.swiftinterface \
+// RUN:   %t/LocalSystemModule.swift -I %t \
+// RUN:   -enable-testing
+
+/// Build a test client once with indexing.
+// RUN: %target-swift-frontend -typecheck %t/TestClient.swift \
+// RUN:   -I %t -module-cache-path %t/cache \
+// RUN:   -index-system-modules \
+// RUN:   -index-store-path %t/idx \
+// RUN:   -index-ignore-stdlib \
+// RUN:   -Rmodule-loading 2>&1 \
+// RUN:   | %FileCheck -check-prefix=CHECK-SWIFTMODULE %s
+// CHECK-SWIFTMODULE: source: '{{.*}}LocalSystemModule.swiftmodule'
+
+/// Build the test client again, this one should still use the adjacent
+/// module that is built for testing.
+// RUN: %target-swift-frontend -typecheck %t/TestClient.swift \
+// RUN:   -I %t -module-cache-path %t/cache \
+// RUN:   -Rmodule-loading 2>&1 \
+// RUN:   | %FileCheck -check-prefix=CHECK-SWIFTMODULE %s
+
+/// @testable import of a module in the SDK works once but not after it's cached.
```

Comment:
I'm okay with only indexing SDK modules (@akyrtzi @benlangmuir in case you have objections), but I do wonder if maybe `@testable` should never be using the module cache here.

Reply:
It's somewhat difficult to change the loading behavior depending on how it's imported because the same module may be imported both normally and `@testable` by different source files, or transitively by another dependency. By the time we encounter the `@testable` import the module could have already been loaded from the cache.